simulation/app.py

```diff
@@ -0,0 +1,621 @@
+    })
+
+    return fig
+
+
+@app.callback(Output('move-population', 'figure'),
+              [Input('interval-component', 'n_intervals'),
+               Input('simulation-data', 'data')])
+def update_graph_move(n_intervals: int, json_data: json) -> 'plotly.express.scatter':
+    """
+    Function handling scatter graph for each individual instance move in
+    grid.
+
+    Parameters
+    ---------
+    n_intervals: int, required
+        Number of passed intervals from begin of simulation. Generated
+        with dash object dcc.Interval.
+    json_data: json, required
+        JSON contain data of simulation.
+
+    Returns
+    -------
+    Return plolty express scatter chart with placed instances in grid.
+    """
+
+    data = json.loads(json_data)
+    cords = {'x': [], 'y': [], 'condition': []}
+
+    for element in data['individuals']:
+        cords['x'].append(element['x'])
+        cords['y'].append(element['y'])
+        cords['condition'].append(element['condition'])
+
+    df2 = pd.DataFrame(data=cords)
+
+    fig = px.scatter(df2, x='x', y='y', color='condition',
+                     color_discrete_map={
+                        'susceptible': 'blue',
+                        'infected': 'red',
+                        'recovered': 'green',
+                        'dead': 'gray'
+                     }, category_orders={'condition': ['susceptible', 'infected',
+                                                       'recovered', 'dead']})
+    fig.update_traces({
+        'marker': {'size': 10}
+    })
+    fig.update_layout({
+        'title': {'text': 'Population move', 'font': {'size': 30}},
+        'xaxis': {'title': 'Width'},
+        'yaxis': {'title': 'Height'},
+        'autosize': False,
+        'width': 800,
+        'height': 800,
+        'template': 'plotly_dark'
+    })
+
+    return fig
+
+
+@app.callback(Output('population-percent', 'figure'),
+              [Input('interval-component', 'n_intervals'),
+               Input('simulation-data', 'data')])
+def update_population_percent(n_intervals: int, json_data: json) -> 'plotly.express.pie':
+    """
+    Function handling pie chart with percent value of each group in population.
+
+    Parameters
+    ----------
+    n_intervals: int, required
+        Number of past intervals.
+    json_data: json, required
+        JSON contain data of simulation.
+        
+    Returns
+    -------
+    Plotly express object pie chart.
+    """
+
+    data = json.loads(json_data)
+    percent = {'condition': [], 'percent_value': []}
+
+    percent['condition'].append('susceptible')
+    percent['percent_value'].append(data['groups']['susceptible'])
+    percent['condition'].append('infected')
+    percent['percent_value'].append(data['groups']['infected'])
+    percent['condition'].append('recovered')
+    percent['percent_value'].append(data['groups']['recovered'])
+    percent['condition'].append('dead')
+    percent['percent_value'].append(data['groups']['dead'])
```

Comment:
If you create `percent` per each iteration, it will be faster to place whole data in one place at one time instead of appending lists.

```suggestion
    percent = {'condition': ['susceptible', 'infected' ...], 'percent_value': [data['groups']['susceptible'], data['groups']['infected'] ...]}
```